src/types.rs

```diff
@@ -241,6 +231,28 @@ macro_rules! key_methods {
                 Ok(ret)
             }
         }
+
+        #[cfg(feature = "serde")]
+        impl serde::Serialize for $i {
+            fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+            where
+                S: serde::Serializer,
+            {
+                serializer.serialize_str(&format!("{:#x}", &self))
```

Comment:
Looks like we need to get rid of the `0x` prefixes to match the to_string impl
```suggestion
                serializer.serialize_str(&format!("{:x}", &self))
```